api/lib/infrastructure/serializers/xml/siecle-parser.js

```diff
@@ -0,0 +1,115 @@
+const { FileValidationError } = require('../../../domain/errors');
+const xml2js = require('xml2js');
+const saxPath = require('saxpath');
+const { isEmpty, isUndefined } = require('lodash');
+
+const NODE_ORGANIZATION_UAI = '/BEE_ELEVES/PARAMETRES';
+const NODES_SCHOOLING_REGISTRATIONS = '/BEE_ELEVES/DONNEES/*/*';
+const ELEVE_ELEMENT = '<ELEVE';
+const STRUCTURE_ELEVE_ELEMENT = '<STRUCTURES_ELEVE';
+const NO_STUDENTS_IMPORTED_FROM_INVALID_FILE = 'Aucun élève n’a pu être importé depuis ce fichier. Vérifiez que le fichier est conforme.';
+const UAI_SIECLE_FILE_NOT_MATCH_ORGANIZATION_UAI = 'Aucun étudiant n’a été importé. L’import n’est pas possible car l’UAI du fichier SIECLE ne correspond pas à celui de votre établissement. En cas de difficulté, contactez support.pix.fr.';
+
+const XMLSchoolingRegistrationSet = require('./xml-schooling-registration-set');
+
+class SiecleParser {
+  constructor(organization, xmlStreamer) {
+    this.organization = organization;
+    this.xmlStreamer = xmlStreamer;
+    this.schoolingRegistrationsSet = new XMLSchoolingRegistrationSet();
+  }
+
+  async parse() {
+
+    await this._checkUAI();
+
+    const schoolingRegistrations = await this._parseStudent();
+
+    return schoolingRegistrations.filter((schoolingRegistration) => !isUndefined(schoolingRegistration.division));
+  }
+
+  async _checkUAI() {
+    const UAIFromSIECLE = await this.xmlStreamer.perform(_extractUAI);
+    const UAIFromUserOrganization = this.organization.externalId;
+
+    if (UAIFromSIECLE !== UAIFromUserOrganization) {
+      throw new FileValidationError(UAI_SIECLE_FILE_NOT_MATCH_ORGANIZATION_UAI);
+    }
+  }
+
+  async _parseStudent() {
+    const bindedExtractMethod = this._extractStudentRegistrationsFromStream.bind(this)
+    return await this.xmlStreamer.perform(bindedExtractMethod);
```

Comment:
Le participe passé de `bind` est `bound` et non `binded`; mais pourquoi ne pas écrire plus directement
```suggestion
    return this.xmlStreamer.perform((stream) => this._extractStudentRegistrationsFromStream(stream));
```
?

Reply:
J'y est pas pensé mais je change pour l'arrow function !